snd-server/services/search_history.py

```diff
@@ -0,0 +1,33 @@
+from databases import Database
+from sqlite3 import IntegrityError
+
+from context import Context
+from models.users import UserInDB
+
+
+class SearchHistory:
+    def __init__(self, ctx: Context):
+        self.db = Database(f"sqlite+aiosqlite:{ctx.config.sqlite_db_path}")
+        self.ctx = ctx
+
+    async def store_track(self, user: UserInDB, track_id: str):
+        user_id = await self.ctx.users_storage.get_user_id(user)
```

Comment:
Can we make `UserInDB` to contain user_id as well? I suppose we are fetching the whole user from db anyway on auth step? Then you don't have to make another db call here